autoload/build.vim

```diff
@@ -328,45 +442,63 @@ endfunction " }}}
 function! build#init(...) " {{{
   if a:0 > 1
     return s:log_err('build#init(): too many arguments. Takes 0 or 1 argument')
+  elseif a:0 ==# 1
+    return build#target('init ' . a:1)
+  else
+    return build#target('init')
   endif
+endfunction " }}}
 
-  let l:build_system = build#get_current_build_system()
+" Print usage examples for the given commands.
+function! s:print_command_examples(commands, build_system) " {{{
+  for [l:subcmd, l:command] in items(a:commands)
+    echo '  :Build ' . l:subcmd . ' [args...]'
+    echo '      => ' . s:prepare_cmd_for_shell(l:command, a:build_system) . ' [args...]'
+  endfor
+endfunction " }}}
 
-  if empty(l:build_system)
-    return s:log('Current file does not belong to any known build system')
-  endif
+" Print an info message for the given build system
+function! s:info_message(build_system)
+  let l:commands = s:gather_commands(a:build_system)
 
-  let l:init_cmd = s:get_buildsys_item(l:build_system.name, 'init')
-  if empty(l:init_cmd)
-    return s:log('Build system "' . l:build_system.name . '" does not need to be initialized')
+  if a:build_system.fallback
+    call s:log('Current file does not belong to any known build system')
+    if empty(l:commands)
+      echo 'No fallback commands defined for filetype "' . &filetype . '".'
+      return
+    endif
+    echo 'Fallback commands are provided. See the examples below.'
+  else
+    echo 'Build system:      ' . a:build_system.name
+    echo 'Project directory: ' . a:build_system.path
   endif
+endfunction
 
-  call s:run_in_env(l:build_system.path, l:init_cmd . (a:0? ' ' . a:1 : ''))
-endfunction " }}}
-
-" Print usage examples for the given fallback targets.
-function! s:print_fallback_examples(fallback_targets) " {{{
-  for [target, command] in items(a:fallback_targets)
-    echo '  :Build ' . target . ' [args...]'
-    echo '      => ' . s:prepare_cmd_for_shell(command) . ' [args...]'
-  endfor
-endfunction " }}}
+" Print a help message for the given build system
+function! s:help_message(build_system)
+  let l:commands = s:gather_commands(a:build_system)
+  echo 'Usage:'
+  echo '  :Build [SUBCMD [args...]]'
+  echo "\n"
+  echo 'Examples:'
+  call s:print_command_examples(l:commands, a:build_system)
+endfunction
```

Comment:
I understand for the command but I think it is still nicer to have separated functions internally because it enables a better help message when using a command that is not available.